src/test/java/com/lithium/dbi/rdbi/recipes/scheduler/DedupJobSchedulerTest.java

```diff
@@ -0,0 +1,318 @@
+            } else if(jobInfo.getJobStr().equals("job5")) {
+                job5Reserved = true;
+            } else if(jobInfo.getJobStr().equals("job6")) {
+                job6Reserved = true;
+            } else {
+                fail("Reserved an unexpected job: " + jobInfo.getJobStr());
+            }
+        }
+        assertTrue(job3Reserved);
+        assertTrue(job5Reserved);
+        assertTrue(job6Reserved);
+    }
+
+    @Test
+    public void testMultiReservationsWithLessAvailableThanLimit() {
+        // Schedule a job
+        assertTrue(scheduledJobSystem.schedule(tubeName, "job1", -5000));
+        // Reserve the job so its in the running queue.
+        scheduledJobSystem.reserveSingle(tubeName, 1000);
+
+        // Schedule another job.
+        assertTrue(scheduledJobSystem.schedule(tubeName, "job2", -5000));
+        // Reserve the job so its in the running queue.
+        scheduledJobSystem.reserveSingle(tubeName, 1000);
+
+        assertTrue(scheduledJobSystem.schedule(tubeName, "job1", -10000));  // Should not be able to reserve since its in running queue
+        assertTrue(scheduledJobSystem.schedule(tubeName, "job3", -6000));   // Should be able to reserve first iteration
+        assertTrue(scheduledJobSystem.schedule(tubeName, "job2", -5000));   // Should not be able to reserve since its in the running queue
+        assertTrue(scheduledJobSystem.schedule(tubeName, "job4", 898789000)); // Not ready to be reserved yet.
+
+        // Reserve 3 jobs. Should return job3 its the next job ready to be reserved.
+        // job1 and job2 can't be reserved since they are the running queue when this reservation request is made.
+        List<TimeJobInfo> reservedJobs = scheduledJobSystem.reserveMulti(tubeName, 1000, 3);
+
+        assertEquals(reservedJobs.size(), 1);
+        assertEquals(reservedJobs.get(0).getJobStr(), "job3");
+    }
+
+    @Test
+    public void testDelete() throws InterruptedException {
+        assertFalse(scheduledJobSystem.deleteJob(tubeName, "{hello:world}"));
+
+        // Schedule and reserve job.
+        scheduledJobSystem.schedule(tubeName, "{hello:world}", 0);
+        JobInfo result1 = scheduledJobSystem.reserveSingle(tubeName, 1000);
+        assertEquals(result1.getJobStr(), "{hello:world}");
+
+        // Delete the job thats in the running queue.
+        assertTrue(scheduledJobSystem.deleteJob(tubeName, result1.getJobStr()));
+
+        // Verify you cant reserve any jobs.
+        JobInfo result2 = scheduledJobSystem.reserveSingle(tubeName, 1000);
+        assertNull(result2);
+
+        // Schedule the job again.
+        scheduledJobSystem.schedule(tubeName, "{hello:world}", 0);
+
+        //Delete job while its in the ready queue.
+        assertTrue(scheduledJobSystem.deleteJob(tubeName, "{hello:world}"));
+
+        // Try to reserve and shouldn't be able to.
+        JobInfo result4 = scheduledJobSystem.reserveSingle(tubeName, 1000);
+        assertNull(result4);
+    }
```

Comment:
you tested each queue (ready & running) individually, but not both together. (i.e. submit, reserve, submit, delete, assertNull(reserve))